example/advection/advection.cpp

```diff
@@ -0,0 +1,399 @@
+        }
+      }
+    }
+  }
+  // get y-fluxes
+  if (pmb->pmy_mesh->ndim>=2) {
+    for (int k=ks; k<=ke; k++) {
+      pmb->precon->DonorCellX2(k, js-1, is, ie, q.data, ql, qr);
+      for (int j=js; j<=je+1; j++) {
+        pmb->precon->DonorCellX2(k, j, is, ie, q.data, qltemp, qr);
+        if (vy > 0.0) {
+          for (int i=is; i<=ie; i++) {
+            q.flux[1](k,j,i) = ql(i)*vy;
+          }
+        } else {
+          for (int i=is; i<=ie; i++) {
+            q.flux[1](k,j,i) = qr(i)*vy;
+          }
+        }
+        auto temp = ql;
+        ql = qltemp;
+        qltemp = temp;
+      }
+    }
+  }
+
+  // TODO(jcd): implement z-fluxes
+
+  return TaskStatus::success;
+
+}
+
+} // namespace Advection
+
+// *************************************************//
+// define the application driver. in this case,    *//
+// that just means defining the MakeTaskList       *//
+// function.                                       *//
+// *************************************************//
+// first some helper tasks
+TaskStatus UpdateContainer(MeshBlock *pmb, int stage, std::vector<std::string>& stage_name, Integrator* integrator) {
+  //const Real beta = stage_wghts[stage-1].beta;
+  const Real beta = integrator->_beta[stage-1];
+  Container<Real>& base = pmb->real_containers.Get();
+  Container<Real>& cin = pmb->real_containers.Get(stage_name[stage-1]);
+  Container<Real>& cout = pmb->real_containers.Get(stage_name[stage]);
+  Container<Real>& dudt = pmb->real_containers.Get("dUdt");
+  parthenon::Update::AverageContainers(cin, base, beta);
+  parthenon::Update::UpdateContainer(cin, dudt, beta*pmb->pmy_mesh->dt, cout);
+  return TaskStatus::success;
+}
+
+
+TaskList AdvectionDriver::MakeTaskList(MeshBlock *pmb, int stage) {
+  TaskList tl;
+  // we're going to populate our last with multiple kinds of tasks
+  // these lambdas just clean up the interface to adding tasks of the relevant kinds
+  auto AddMyTask = [&tl, pmb, stage, this] (BlockStageNamesIntegratorTaskFunc func, TaskID dep) {
+    return tl.AddTask<BlockStageNamesIntegratorTask>(func,dep,pmb,stage,stage_name,integrator);
+  };
+  auto AddContainerTask = [&tl] (ContainerTaskFunc func, TaskID dep, Container<Real>& rc) {
+    return tl.AddTask<ContainerTask>(func,dep,rc);
+  };
+  auto AddTwoContainerTask = [&tl] (TwoContainerTaskFunc func, TaskID dep, Container<Real>& rc1, Container<Real>& rc2) {
+    return tl.AddTask<TwoContainerTask>(func,dep,rc1,rc2);
+  };
+
+  TaskID none(0);
+  // first make other useful containers
+  if (stage == 1) {
+    Container<Real>& base = pmb->real_containers.Get();
+    pmb->real_containers.Add("dUdt", base);
+    for (int i=1; i<integrator->_nstages; i++) pmb->real_containers.Add(stage_name[i], base);
+  }
+
+  Container<Real>& sc0  = pmb->real_containers.Get(stage_name[stage-1]);
+  Container<Real>& sc1  = pmb->real_containers.Get(stage_name[stage]);
+  Container<Real>& dudt = pmb->real_containers.Get("dUdt");
```

Comment:
This could use some explanation too. i.e., that `sc0` is the last subcycle and that `sc1` is the current one. Just to clarify how this works when we're doing multistage integration.

Reply:
added some comments